analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -333,9 +333,9 @@ expression switch
         public static bool IsLeftSideOfAssignment(this ExpressionSyntax expression)
         {
             var topParenthesizedExpression = expression.GetSelfOrTopParenthesizedExpression();
-            return topParenthesizedExpression.Parent.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
-                topParenthesizedExpression.Parent is AssignmentExpressionSyntax assignment &&
-                assignment.Left == topParenthesizedExpression;
+            return topParenthesizedExpression.Parent.IsKind(SyntaxKind.SimpleAssignmentExpression)
+                    && topParenthesizedExpression.Parent is AssignmentExpressionSyntax assignment
```

Comment:
NITPICK: the indentation should be aligned with the start of the topParenthesizedExpression in the line above. (Currently there is 1 space more before both && operators)